docs/source/semantic_segmentation.mdx

```diff
@@ -0,0 +1,161 @@
+
+...    plt.figure(figsize=(15, 10))
+...    plt.imshow(img)
+...    plt.axis("off")
+...    plt.show()
+
+
+>>> visualize_seg_mask(
+...     np.array(dataset["train"][index]["image"]),
+...     np.array(dataset["train"][index]["annotation"])
+... )
+```
+
+<div class="flex justify-center">
+    <img src="https://i.imgur.com/QoQ9R0f.png">
+</div>
+
+Now apply some augmentations with `albumentations`. You’ll first resize the image and adjust its brightness.
+
+```py
+>>> import albumentations as A
+
+>>> transform = A.Compose(
+...     [
+...         A.Resize(256, 256),
+...         A.RandomBrightnessContrast(brightness_limit=0.3, contrast_limit=0.3, p=0.5),
+...     ]
+... )
+```
+
+Create a function to apply the transformation to the images:
+
+```py
+>>> def transforms(examples):
+...     transformed_images, transformed_masks = [], []
+...
+...     for image, seg_mask in zip(examples["image"], examples["annotation"]):
+...         image, seg_mask = np.array(image), np.array(seg_mask)
+...         transformed = transform(image=image, mask=seg_mask)
+...         transformed_images.append(transformed["image"])
+...         transformed_masks.append(transformed["mask"])
+...
+...     examples["pixel_values"] = transformed_images
+...     examples["label"] = transformed_masks
+...     return examples
+```
+
+Use the [`~Dataset.set_transform`] function to apply the transformation on-the-fly to batches of the dataset to consume less disk space:
+
+```py
+>>> dataset.set_transform(transforms)
+```
+
+You can verify the transformation worked by indexing into the `pixel_values` and `label` of the `index`th example:
+
+```py
+>>> image = np.array(dataset["train"][index]["pixel_values"])
+>>> mask = np.array(dataset["train"][index]["label"])
+
+>>> visualize_seg_mask(image, mask)
+```
+
+<div class="flex justify-center">
+    <img src="https://i.imgur.com/xCU11KU.png">
+</div>
+
+In this guide, you have used `albumentations` for augmenting the dataset. It's also possible to use `torchvision` for that purpose. 
+
+```py 
+>>> from torchvision.transforms import Resize, ColorJitter, Compose
+
+>>> transformation_chain = Compose([
+...     Resize((256, 256)),
+...     ColorJitter(brightness=0.25, contrast=0.25, saturation=0.25, hue=0.1)
+... ])
+>>> resize = Resize((256, 256))
+
+>>> def train_transforms(example_batch):
+...     example_batch["pixel_values"] = [transformation_chain(x) for x in example_batch["image"]]
+...     example_batch["label"] = [resize(x) for x in example_batch["annotation"]]
+...     return example_batch
+
+>>> dataset.set_transform(train_transforms)
+
+>>> image = np.array(dataset["train"][index]["pixel_values"])
+>>> mask = np.array(dataset["train"][index]["label"])
+
+>>> visualize_seg_mask(image, mask)
+```
+
+<div class="flex justify-center">
+    <img src="https://i.imgur.com/9Jwn0mN.png">
+</div>
+
+You can follow [this resource](https://huggingface.co/docs/transformers/tasks/semantic_segmentation)
```

Comment:
I think the transition here can be a little smoother. Maybe something like:

Now that you know how to process a dataset for semantic segmentation, learn [how to train a semantic segmentation model](link to doc) and use it for inference.

Reply:
I agree it's a better transition. Maybe we could follow something similar for the rest of the related guides in `datasets`?

Reply:
For sure, that sounds good to me :)